aiokafka/group_coordinator.py

```diff
@@ -0,0 +1,847 @@
+                self.group_id,
+                list(topic_partitions.items())
+            )
+
+            try:
+                offsets = yield from self._proc_offsets_fetch_request(
+                    node_id, request)
+            except Errors.KafkaError as err:
+                if not err.retriable:
+                    raise err
+                else:
+                    # wait backoff and try again
+                    yield from asyncio.sleep(
+                        self._retry_backoff_ms / 1000.0, loop=self.loop)
+            else:
+                return offsets
+
+    @asyncio.coroutine
+    def _proc_offsets_fetch_request(self, node_id, request):
+        response = yield from self._send_req(node_id, request)
+        offsets = {}
+        for topic, partitions in response.topics:
+            for partition, offset, metadata, error_code in partitions:
+                tp = TopicPartition(topic, partition)
+                error_type = Errors.for_code(error_code)
+                if error_type is not Errors.NoError:
+                    error = error_type()
+                    log.debug("Error fetching offset for %s: %s", tp, error)
+                    if error_type is Errors.GroupLoadInProgressError:
+                        # just retry
+                        raise error
+                    elif error_type is Errors.NotCoordinatorForGroupError:
+                        # re-discover the coordinator and retry
+                        self.coordinator_dead()
+                        raise error
+                    elif error_type in (Errors.UnknownMemberIdError,
+                                        Errors.IllegalGenerationError):
+                        # need to re-join group
+                        self._subscription.mark_for_reassignment()
+                        raise error
+                    elif error_type is Errors.UnknownTopicOrPartitionError:
+                        log.warning(
+                            "OffsetFetchRequest -- unknown topic %s", topic)
+                        continue
+                    else:
+                        log.error("Unknown error fetching offsets for %s: %s",
+                                  tp, error)
+                        raise error
+                elif offset >= 0:
+                    # record the position with the offset
+                    # (-1 indicates no committed offset to fetch)
+                    offsets[tp] = OffsetAndMetadata(offset, metadata)
+                else:
+                    log.debug(
+                        "No committed offset for partition %s", tp)
+
+        return offsets
+
+    @asyncio.coroutine
+    def commit_offsets(self, offsets):
+        """Commit specific offsets asynchronously.
+
+        Arguments:
+            offsets (dict {TopicPartition: OffsetAndMetadata}): what to commit
+
+        Raises error on failure
+        """
+        assert all(map(lambda k: isinstance(k, TopicPartition), offsets))
```

Comment:
I think it is enough to do this check on consumer. Coordinator is protected interface anyway